service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,102 @@
+                            <date-filter/><econdition field-name="facilityId"/><econdition field-name="replacesContactMechId" from="contactMechId"/><select-field field-name="contactMechId"/><order-by field-name="contactMechId"/>
+                        </entity-find>
+                        <if condition="replacesFcmiList">
+                            <set field="contactMechId" from="replacesFcmiList[0].contactMechId"/><return/>
+                        </if>
+                    </else-if>
+                </if>
+            </if>
+
+            <set field="path" value="rest/AV"/>
+            <set field="method" value="POST"/>
+            <entity-find-one entity-name="ups.ShippingGatewayUps" value-field="ups">
+                <field-map field-name="shippingGatewayConfigId"/>
+            </entity-find-one>
+            <set field="accessRequestMap" from="[AccessLicenseNumber:ups.accessLicenceNumber, UserId:ups.userId, Password:ups.password]"/>
+
+            <!-- Preparing address map by calling get#AddressMap service-->
+            <service-call name="mantle.ups.UpsServices.get#AddressMap" out-map="addrOut" in-map="[postalAddress:postalAddress, partyId:partyId]"/>
+
+            <set field="addressMap" from="addrOut.addressMap"/>
+            <set field="addressValidationRequestMap" from="[Request:[TransactionReference:[CustomerContext:'Ups Shipping Address Validation'], RequestAction:'AV'], Address:addressMap]"/>
+            <set field="inMap" from="[AccessRequest:accessRequestMap, AddressValidationRequest:addressValidationRequestMap]"/>
+
+            <!-- Calling generic service for sending request  -->
+            <service-call name="mantle.ups.UpsServices.send#UpsRequest" in-map="[shippingGatewayConfigId:shippingGatewayConfigId, path:path, methodType:method, requestMap:inMap]" out-map="addrValidationResponseOut"/>
+            <set field="responseStatusCode" from="addrValidationResponseOut.responseMap.AddressValidationResponse.Response?.ResponseStatusCode"/>
+            <set field="responseStatusDescription" from="addrValidationResponseOut.responseMap.AddressValidationResponse.Response?.ResponseStatusDescription"/>
+
+            <!-- Check if the response is valid or not -->
+            <set field="is_valid" from="'1'.equals(addrValidationResponseOut.responseMap.AddressValidationResponse.Response?.ResponseStatusCode)"/>
+            <!--if responseStatusCode is 1 then only code is giving Success rest in all cases it is Failure: based on that set value of trustLevelEnumId -->
+            <set field="trustLevelEnumId" from="(is_valid) ? 'CmtlValid' : 'CmtlInvalid'"/>
+
+            <log message="Ups validate address ${contactMechId} complete? ${responseStatusDescription} valid? ${responseStatusCode} trust ${trustLevelEnumId}}"/>
+
+            <!--Note: currently service is validating the address only on the basis of the response description that it is either Success or Faliure -->
```

Comment:
I think in the comment it should be on the basis of response status code instead of description.